notebooks/code.py

```diff
@@ -0,0 +1,27 @@
+import dataiku
+
+def list_notebooks(client=None):
+    """Prints all jupyter notebooks in all projects, distinguishes which notebooks are active as well as the session data for active notebooks
+    """ 
+    projects = client.list_projects()
+
+    for p in projects :
+        proj = client.get_project(p["projectKey"])
+        all_notebooks = proj.list_jupyter_notebooks(active=False, as_type="listitems")
```

Comment:
You can avoid calling `list_jupyter_notebooks()` twice 🙂 

  Run it like this **once**: 
```python
list_jupyter_notebooks(active=False, as_type="object") # default function signature
```
and for each resulting `dataikuapi.dss.notebook.DSSJupyterNotebook` object, check if the notebook is active by getting the kernel info using `get_sessions()`.